02-peer-protocol.md

```diff
@@ -1106,6 +1118,71 @@ it's simplest to only allow it to set fee levels; however, as the same
 fee rate applies to HTLC transactions, the receiving node must also
 care about the reasonableness of the fee.
 
+### Assigning Random `short_channel_id`
+
+To facilitate privacy of unannounced channels, peers can request a random
+short_channel_id to use which does not have any relationship to the actual
+channel funding transaction.  Note that these messages are all even, so it
+is illegal to send them to peers which do not offer `option_scid_assign`.
+
+1. type: 260 (`assign_scid`) (`option_scid_assign`)
+2. data:
+   * [`channel_id`:`channel_id`]
+
+1. type: 268 (`assign_scid_reply`) (`option_scid_assign`)
+   * [`channel_id`:`channel_id`]
+   * [`short_channel_id`:`short_channel_id`]
+
+1. type: 270 (`unassign_scid`) (`option_scid_assign`)
+2. data:
+   * [`channel_id`:`channel_id`]
+
+1. type: 272 (`unassign_scid_reply`) (`option_scid_assign`)
+   * [`channel_id`:`channel_id`]
+
+#### Requirements
+
+A sending node:
+  - if it did not offer `option_scid_assign`:
+    - MUST NOT send `assign_scid` or `unassign_scid`
+  - SHOULD remember the last `assign_scid_reply` `short_channel_id` for use in
+    receiving payments.
+  - MAY send `unassign_scid` when no invoices are pending, and send `assign_scid` when
+    it wants to create a new invoice and none are pending.
+
+A receiving node:
+  - on receiving `assign_scid`:
+    - MUST assign an unpredictable unique 64-bit number to this channel
+      (see [Assigned `short_channel_id` Forwarding](04-onion-routing.md#assigned-short_channel_id-forwarding)).
+	  and return it in an ``assign_scid_reply` message.
+	  - SHOULD ensure this does not conflict with future `short_channel_id`s.
+    - MUST discard any previous assigned `short_channel_id` for this channel
```

Comment:
With this requirement, I don't think we need the `unassign_*` messages. If they care about privacy, then they will request new scids regularly. (NB: with this proposal we can't have two pending invoices simultaneously with two different scids, I think that's fine).

Reply:
I think we still want unassign to reduce the chance for probing when you're offline?